packages/pipeline/src/scripts/pull_ddex_orderbook_snapshots.ts

```diff
@@ -0,0 +1,62 @@
+// tslint:disable:no-console
+import { BigNumber } from '@0x/utils';
+import axios from 'axios';
+import * as R from 'ramda';
+import { Connection, ConnectionOptions, createConnection, EntityManager } from 'typeorm';
+
+import { DdexActiveMarketsResponse, DdexMarket, DdexOrder, DdexOrderbookResponse } from '../data_sources/ddex';
+import { TokenOrderbookSnapshot as TokenOrder } from '../entities';
+import * as ormConfig from '../ormconfig';
+import { parseDdexOrders } from '../parsers/ddex_orders';
+import { handleError } from '../utils';
+
+const DDEX_BASE_URL = 'https://api.ddex.io/v2';
+const ACTIVE_MARKETS_URL = DDEX_BASE_URL + '/markets';
+const ORDERBOOK_ENDPOINT = '/orderbook?level=3';
+const DDEX_SOURCE = 'ddex';
+
+let connection: Connection;
+
+(async () => {
+    connection = await createConnection(ormConfig as ConnectionOptions);
+    const markets = await getActiveMarkets();
+    await Promise.all(markets.map(getMarketOrderbook));
+    process.exit(0);
+})().catch(handleError);
+
+/**
+ * Call Ddex API to find out which markets they are maintaining orderbooks for.
+ */
+async function getActiveMarkets(): Promise<DdexMarket[]> {
+    console.log('Getting all active DDEX markets');
+    const resp = await axios.get<DdexActiveMarketsResponse>(ACTIVE_MARKETS_URL);
+    const markets = resp.data.data.markets;
+    console.log(`Got ${markets.length} markets.`);
+    return markets;
+}
+
+/**
+ * Retrieve orderbook from Ddex API for a given market. Parse orders and insert
+ * them into our database.
+ * @param market Object from Ddex API containing market data.
+ */
+async function getMarketOrderbook(market: DdexMarket): Promise<void> {
+    console.log(`${market.id}: Retrieving orderbook.`);
+    const marketOrderbookUrl = ACTIVE_MARKETS_URL + '/' + market.id + ORDERBOOK_ENDPOINT;
+    const resp = await axios.get<DdexOrderbookResponse>(marketOrderbookUrl);
+    const retrievedTimestamp = new BigNumber(Date.now());
+
+    console.log(`${market.id}: Parsing orders.`);
+    const orders = parseDdexOrders(resp.data.data.orderBook, market, retrievedTimestamp, DDEX_SOURCE);
+
+    if (orders.length > 0) {
+        console.log(`${market.id}: Saving ${orders.length} orders.`);
+        await connection.transaction(async (manager: EntityManager): Promise<void> => {
```

Comment:
If you just want to save the orders, you don't need to create a transaction. You can save some code by doing it this way:

```
ordersRepository.save(orders, { chunk: Math.ceil(blocks.length / BATCH_SAVE_SIZE) });
```

Where `BATCH_SAVE_SIZE` is the number of orders to save per query. Batching saves this way also helps us avoid issues with transactions that are too big for Postgres.

See https://github.com/typeorm/typeorm/blob/master/src/repository/SaveOptions.ts#L30